packages/medusa-plugin-typesense/src/services/typesense.ts

```diff
@@ -0,0 +1,108 @@
+import { SearchTypes } from "@medusajs/types"
+import { SearchUtils } from "@medusajs/utils"
+import { Client as Typesense } from "typesense"
+import { CollectionCreateSchema } from "typesense/lib/Typesense/Collections"
+import { TypesensePluginOptions } from "../types"
+import { transformProduct } from "../utils/transformer"
+
+class TypesenseSearchService extends SearchUtils.AbstractSearchService {
+  isDefault = false
+
+  protected readonly client_: Typesense
+  protected readonly config_: TypesensePluginOptions
+
+  constructor(_, options: TypesensePluginOptions) {
+    super(_, options)
+
+    this.config_ = options
+
+    if (process.env.NODE_ENV !== "development") {
+      if (!options.config?.apiKey) {
+        throw Error("Typesense API key is missing in plugin config.")
+      }
+    }
+
+    if (!options.config.nodes?.length) {
+      throw Error("Typesense nodes config is missing in plugin config.")
+    }
+
+    // options.config should be the same format used to initialize the Typesense Client
+    //  as described here: https://typesense.org/docs/0.22.1/api/authentication.html
+    this.client_ = new Typesense(options.config)
+  }
+
+  async createIndex(indexName: string, options: Record<string, unknown>) {
+    // options.collectionSchemas.{indexName} should be the schema of the collection
+    //  as described here: https://typesense.org/docs/0.22.1/api/collections.html#create-a-collection
```

Comment:
I would recommend changing this URL to `https://typesense.org/docs/latest/api/collections.html#create-a-collection`

(Use the latest tag, instead of a specific version)